### PR TITLE
Pay 3947 payment cve fixes

### PR DESCRIPTION
**Pull request by Jennings-HMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: cve-resolution-strategy.gradle [hunk 1]
```diff
@@ -0,0 +1,89 @@
+configurations.all {
+    resolutionStrategy {
+        eachDependency { DependencyResolveDetails det ->
+            if (det.requested.name == 'tomcat-embed-core') {
```
Comment (amitchidrewar1301) [inline cve-resolution-strategy.gradle:4]: As discussed please do the clustering of conditions and add appropriate comments.

File: dependency-check-suppressions.xml [hunk 2]
```diff
@@ -1,111 +1,53 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.3.xsd">
-    <suppress>
-        <notes>CVE-2018-1258: we do not use Spring Method Security, so this does not affect us. see https://pivotal.io/security/cve-2018-1258.
-            we are on latest Spring Framework version : 5.2.2.RELEASE : https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Versions</notes>
+    <suppress until="2021-01-25">
```
Comment (amitchidrewar1301) [inline dependency-check-suppressions.xml:3]: As discussed please do the clustering of conditions and add appropriate comments.

**Review by amitchidrewar1301 — APPROVED** [review]

File: dependency-check-suppressions.xml (hunk 2 above)
Comment (Jennings-HMCTS) [reply dependency-check-suppressions.xml:5]: supressions are grouped and comments are added

File: cve-resolution-strategy.gradle (hunk 1 above)
Comment (Jennings-HMCTS) [reply cve-resolution-strategy.gradle:10]: fixes are grouped and added the comments

**Review by krishnanuthalapati — APPROVED** [review]

**Pull request merged**